Shared/Samples/Run valve isolation trace/RunValveIsolationTraceView.Model.swift

```diff
@@ -0,0 +1,402 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import Combine
+import Foundation
+
+extension RunValveIsolationTraceView {
+    /// The view model for this sample.
+    @MainActor
+    class Model: ObservableObject {
+        /// A map with a 'night time' street map style.
+        let map = Map(basemapStyle: .arcGISStreetsNight)
+        
+        /// The utility network for this sample.
+        private let utilityNetwork = UtilityNetwork(url: .featureServiceURL)
+        
+        /// The service geodatabase used to create the feature layer.
+        private let serviceGeodatabase = ServiceGeodatabase(url: .featureServiceURL)
+        
+        /// The service geodatabase feature layers.
+        private var layers: [FeatureLayer] = []
+        
+        /// The last element that was added to the list of filter barriers.
+        ///
+        /// When an element contains more than one terminal, the user should be presented with the
+        /// option to select a terminal. Keeping a reference to the last added element provides ease
+        /// of access to save the user's choice.
+        @Published private(set) var lastAddedElement: UtilityElement?
+        
+        /// The current tracing related activity.
+        @Published private(set) var tracingActivity: TracingActivity? = .loadingServiceGeodatabase
+        
+        /// The base trace parameters.
+        private var traceParameters = UtilityTraceParameters(traceType: .isolation, startingLocations: [])
+        
+        /// The point geometry of the starting location.
+        private(set) var startingLocationPoint: Point!
```

Comment:
This shouldn't be IUO. If the model setup method fails silently, the `setViewpointCenter` method in the view will crash.

Reply:
If the point does fail to unwrap should a custom error be thrown or fatalError?

Reply:
IMO, It's not good for samples to crash if data changes.

Reply:
There is already a mechanism in this sample to show errors in a status text:

```swift

        /// Performs important tasks including adding credentials, loading and adding operational layers.
        func setup() async {
            do {
                try await ArcGISEnvironment.authenticationManager.arcGISCredentialStore.add(.publicSample)
                try await loadServiceGeodatabase()
                try await loadUtilityNetwork()
            } catch {
                statusText = error.localizedDescription
            }
        }
```

If you simply throw an error it will get displayed.

Reply:
Fixed.